sdk/program/src/sysvar/slot_hashes.rs

```diff
@@ -62,13 +72,112 @@ impl Sysvar for SlotHashes {
+        }
+        Ok(pod_hashes
+            .binary_search_by(|PodSlotHash { slot: this, .. }| slot.cmp(this))
+            .map(|idx| pod_hashes[idx].hash)
+            .ok())
+    }
+
+    /// Get the position of an entry in the sysvar by its key.
+    /// Returns `None` if the key is not found.
+    pub fn position(slot: &Slot) -> Result<Option<usize>, ProgramError> {
+        let mut pod_hashes = vec![PodSlotHash::default(); MAX_ENTRIES];
+        {
+            let data = bytemuck::try_cast_slice_mut::<PodSlotHash, u8>(&mut pod_hashes)
+                .map_err(|_| ProgramError::InvalidAccountData)?;
+
+            // Ensure the created buffer is aligned to 8.
+            if data.as_ptr().align_offset(8) != 0 {
+                return Err(ProgramError::InvalidAccountData);
+            }
+
+            let offset = 8; // Vector length as `u64`.
+            let length = (SlotHashes::size_of() as u64).saturating_sub(offset);
+            get_sysvar(data, &SlotHashes::id(), offset, length)?;
+        }
+        Ok(pod_hashes
+            .binary_search_by(|PodSlotHash { slot: this, .. }| slot.cmp(this))
+            .ok())
+    }
+}
+
 #[cfg(test)]
 mod tests {
     use {
         super::*,
-        crate::{clock::Slot, hash::Hash, slot_hashes::MAX_ENTRIES},
+        crate::{
+            clock::Slot,
+            entrypoint::SUCCESS,
+            hash::{hash, Hash},
+            program_stubs::{set_syscall_stubs, SyscallStubs},
+            slot_hashes::{SlotHash, MAX_ENTRIES},
+        },
     };
 
+    struct MockSlotHashesSyscall {
+        slot_hashes: SlotHashes,
+    }
+
+    impl SyscallStubs for MockSlotHashesSyscall {
+        #[allow(clippy::arithmetic_side_effects)]
+        fn sol_get_sysvar(
+            &self,
+            _sysvar_id_addr: *const u8,
+            var_addr: *mut u8,
+            offset: u64,
+            length: u64,
+        ) -> u64 {
+            // The syscall tests for `sol_get_sysvar` should ensure the following:
+            //
+            // - The provided `sysvar_id_addr` can be translated into a valid
+            //   sysvar ID for a sysvar contained in the sysvar cache, of which
+            //   `SlotHashes` is one.
+            // - Length and memory checks on `offset` and `length`.
+            //
+            // Therefore this mockup can simply just unsafely use the provided
+            // `offset` and `length` to copy the serialized `SlotHashes` into
+            // the provided `var_addr`.
+            let data = bincode::serialize(&self.slot_hashes).unwrap();
+            let slice = unsafe { std::slice::from_raw_parts_mut(var_addr, length as usize) };
+            slice.copy_from_slice(&data[offset as usize..(offset + length) as usize]);
+            SUCCESS
+        }
+    }
+
+    fn mock_get_sysvar_syscall(slot_hashes: &[SlotHash]) {
+        set_syscall_stubs(Box::new(MockSlotHashesSyscall {
+            slot_hashes: SlotHashes::new(slot_hashes),
+        }));
+    }
```

Comment:
nit: this isn't needed since there's only one test, but to be safe for the future in case we add more tests, could we wrap this test in a `Once`? https://doc.rust-lang.org/std/sync/struct.Once.html

```suggestion
    fn mock_get_sysvar_syscall(slot_hashes: &[SlotHash]) {
        static ONCE: std::sync::Once = std::sync::Once::new();
        ONCE.call_once(|| {
            set_syscall_stubs(Box::new(MockSlotHashesSyscall {
                slot_hashes: SlotHashes::new(slot_hashes),
            }));
        });
    }
```